scripts/commands/checkLinks.ts

```diff
@@ -132,7 +139,8 @@ async function loadFilesAndLinks(
 function loadFiles(existingPaths: string[]): File[] {
   const fileList: File[] = [];
   for (let path of existingPaths) {
-    fileList.push(new File(path, []));
+    const synthetic = SYNTHETIC_FILES.includes(path) ? true : false;
+    fileList.push(new File(path, [], synthetic));
```

Comment:
Hm, I'm realizing this function doesn't make a ton of sense to me. It creates `File` objects, but they're wrong! They are claiming there are no anchors when some of them do have it.

We're not mutating `File` objects with something like `setAnchors`, which is great. So, these `File` objects aren't very useful.

What would it look like to remove this function?

--

One idea to explore is making `SYNTHETIC_FILES` be a `File[]` rather than `string[]`. You could do this:

```typescript
["my-file1", "file2"].map((fp) => new File(fp, [], true))
```

Reply:
I like the idea of stopping using a function to create "false" files. I have changed the code to use your suggestion, creating an array of extra files once we have parsed the files we want (line 144).